Adding two examples (#265)
discordjs/guide: guide/popular-topics/common-questions.md

}
```

> ### How do I check which role was added/removed, and for which member?

Reviewer: > Furthermore, the question is more of a miscellaneous question than an administrative one.

I agree, this should not be placed in the Administrative section but misc (subheading of common questions, not misc. examples)

<!-- eslint-skip -->

```js
// We start by declaring a guildMemberUpdate listener

> // This should ***not*** be inside the message listener (or any other listener)

Reviewer: I don't think this reads very well, I'd suggest something along the lines of:
```suggestion
// This code should be placed outside of any other listener callbacks to prevent listener nesting
```

<client>.on('guildMemberUpdate', (oldMember, newMember) => {

> 	if(oldMember.roles.size === newMember.roles.size) return;

Reviewer: This suggests that roles can only be changed if the size of the role collection changes, which is wrong.
Roles can not only be added or removed, and not only one by one either (relevant further down for the usage of `find`). 

The approach I tend to suggest to people with this request is the following:
is in old but not in new => removed
is in new but not in old =>added
(is in both => unchanged) which i'd not include here

### How do I find all online members?

Assuming the process is to be done for the guild the message is sent in.
<!-- eslint-skip -->

```js
// First we use fetchMembers to make sure all members are cached
<message>.guild.fetchMembers().then(fetchedGuild => {
	const totalOnline = fetchedGuild.members.filter(member => member.presence.status === 'online');
	// We now have an array with all online member objects in the totalOnline variable
	<message>.channel.send(`There are currently ${totalOnline.size} members online in this guild!`);
});
```

### How do I check which role was added/removed, and for which member?

<!-- eslint-skip -->

```js
// We start by declaring a guildMemberUpdate listener
// This code should be placed outside of any other listener callbacks to prevent listener nesting

<client>.on('guildMemberUpdate', (oldMember, newMember) => {

> 	const removedRoles = oldMember.roles.filter(role => !newMember.roles.some(roleTwo => roleTwo.name === role.name));

Reviewer: Not a fan of filtering by name here, id is way more robust against edge cases
```suggestion
	const removedRoles = oldMember.roles.filter(role => !newMember.roles.has(role.id));
```

	if(removedRoles.size > 0) console.log(`The roles ${removedRoles.map(r => r.name)} were added to ${oldMember.displayName}.`);

> 	const addedRoles = newMember.roles.filter(role => !oldMember.roles.some(roleTwo => roleTwo.name === role.name));

Reviewer: same as above
```suggestion
	const addedRoles = newMember.roles.filter(role => !oldMember.roles.has(role.id));
```

### What is the difference between a User and a GuildMember?

A lot of users get confused as to what the difference between Users and GuildMembers is. The simple answer is that a User represents a global Discord user and a GuildMember represents a Discord user on a specific server. That means only GuildMembers can have permissions, roles, and nicknames, for example, because all of these things are server-bound information that could be different on each server that user is in.

### How do I find all online members?

Assuming the process is to be done for the guild the message is sent in.
<!-- eslint-skip -->

```js
// First we use fetchMembers to make sure all members are cached
<message>.guild.fetchMembers().then(fetchedGuild => {
	const totalOnline = fetchedGuild.members.filter(member => member.presence.status === 'online');

> 	// We now have an array with all online member objects in the totalOnline variable

Reviewer 2: ```suggestion
	// We now have a collection with all online member objects in the totalOnline variable
```
[`Collection#filter`](https://discord.js.org/#/docs/main/stable/class/Collection?scrollTo=filter) returns a Collection.

### What is the difference between a User and a GuildMember?

A lot of users get confused as to what the difference between Users and GuildMembers is. The simple answer is that a User represents a global Discord user and a GuildMember represents a Discord user on a specific server. That means only GuildMembers can have permissions, roles, and nicknames, for example, because all of these things are server-bound information that could be different on each server that user is in.

### How do I find all online members?

Assuming the process is to be done for the guild the message is sent in.
<!-- eslint-skip -->

```js

> // First we fetchMembers to be 100% sure everyone is in the member cache

Reviewer: I don't think this reads very well, I'd suggest something along the lines of:
```suggestion
// First we use fetchMembers to make sure all members are cached
```
